vsifile/io/base.py

```diff
@@ -70,34 +71,36 @@ def __hash__(self):
         ...
 
     def _get_header(self) -> bytes:
-        logger.debug("VSIFILE_INFO: HEAD (Open)")
-        head = obs.head(self._store, self._key)
-        self._size = head["size"]
-
-        header = self.header_cache.get(f"{self.name}-header", read=True)
-        if not header:
-            end = (
-                vsi_settings.ingested_bytes_at_open
-                if self._size > vsi_settings.ingested_bytes_at_open
-                else self._size
-            )
+        cache = self.header_cache.get(f"{self.name}-header", read=True)
+        if not cache:
             logger.debug("VSIFILE_INFO: GET")
-            logger.debug(f"VSIFILE: Downloading: 0-{end}")
-            header = bytes(obs.get_range(self._store, self._key, start=0, end=end))
+            logger.debug(f"VSIFILE: Downloading: 0-{vsi_settings.ingested_bytes_at_open}")
```

Comment:
in theory we could end up downloading less than `ingested_bytes_at_open` but I don't want to add complexity to the code to return the exact size that was downloaded (user could check `len(VSIFile().header)`)